src/main/kotlin/org/rust/cargo/toolchain/tools/Rustup.kt

```diff
@@ -43,12 +45,28 @@ class Rustup(toolchain: RsToolchain, private val projectDirectory: Path) : RsToo
         }
     }
 
+    data class Target(val name: String, val isInstalled: Boolean) {
+        companion object {
+            fun from(line: String): Target {
+                val name = line.substringBefore(' ')
+                val isInstalled = line.substringAfter(' ') == "(installed)"
```

Comment:
Hmm, I'm not sure why we check `"(default)"` case in `Component#from`, and I haven't managed to simulate such output from `rustup component list` and `rustup target list` (but such output is definetely possible for `rustup toolchain list`).

@mchernyavsky Could you please explain is it really necessary?

Reply:
I think this is needed on older rustup versions

(see the output from this issue - https://github.com/rust-lang/rustup/issues/1544)

Reply:
@mchernyavsky Thank you! So we added `"(default)"` case for targets as well just in case